src/lasp_dependence_dag.erl

```diff
@@ -44,20 +53,53 @@
 %% lets us quickly delete those edges.
 -type process_map() :: dict:dict(pid(), {id(), id()}).
 
+%% Stored metadata for a lasp process.
+%% We don't store the input(s) and output vertices as that information
+%% is implicitly stored by edges in the graph.
+-record(process_metadata, {read :: function(),
+                           transform :: function(),
+                           write :: function()}).
+
+%% We store a mapping Id -> {Pid, Path, MetadataList} to identify the list of
+%% vertices and relationships that get removed as part of a path contraction.
+%%
+%% Used during the cleaving step.
+-type contract_process_id() :: non_neg_integer().
```

Comment:
Make define a type for the hash and comment it to say how it's computed.  It's not straightforward in the code, but is one you explain it.